feat: create vben project

wu-clan opened the pull request.

wu-clan commented:
please review.

downdawn commented:
What formatting tool are you using?
Actually, the Arco project was initialized with Husky configuration, but I removed it for convenience when committing.

wu-clan commented:
Yes, husky for git

The formatting of this PR is just a simple formatting based on webstom, including optimized import, sorting (alphabetical order, empty lines of code), no specific coding specification

wu-clan commented:
The final specification is eslint

wu-clan commented:
We can pick up husky again

downdawn commented:
OK

downdawn commented on src/hooks/user.ts:
```diff
@@ -10,8 +10,8 @@ export default function useUser() {
     const currentRoute = router.currentRoute.value;
     Message.success('登出成功');
-    router.push({
-      name: logoutTo && typeof logoutTo === 'string' ? logoutTo : 'login',
```
Here it seems that it cannot be reduced to True

wu-clan replied:
ts should not need to judge the type.

downdawn commented:
LGTM.

vince292007 committed:
chore: update eslint configuration (#4872)

ArthurDarkstone committed:
feat: add reize components & demo (#4862)

* feat: resize component

* chore: change positon of resize components

* feat: add resize demo

* chore: resize demo completed

* chore: fix display number

* chore: add infer comment

* fix: move reszie demo to examples

* fix: fix icon & removed scss

anncwb committed:
fix: resolve issue with grid reload parameter not working (#4888)

anncwb committed:
fix: correct grid styling issue (#4889)

anncwb committed:
fix: resolve onChange issue in form component (#4890)

wdzyy committed:
fix(@vben/common-ui): pagination current page error (#4893)

anncwb committed:
chore: release v5.4.7

mynetfan committed:
fix: adjust `useWatermark` logic (#4896)

Mintnoii committed:
docs: optimize the introduction in both Chinese and English (#4913)

* 优化简介中文文档

1. 优化文案及病句
2. 统一格式

* Optimize the introduction English document

1. Optimize copywriting and sentences
2. Uniform format

mynetfan committed:
fix: dialog `opened/closed` event triggered incorrectly,fixed #4902 (#4908)

ArthurDarkstone committed:
feat: add 3 resize examples (#4907)

DesignHhuang committed:
fix: fieldMappingTime data error when clear inputvalue (#4906)

ryomahan committed:
fix: form-api.setValues can't resolve nested fields (#4915)

fix #4912

anncwb committed:
fix: resolve issue with Upload component not working correctly inside Form (#4916)

anncwb committed:
fix: supplement datepicker component (#4943)

* fix: supplement datepicker component

* chore: typo

anncwb committed:
chore: release 5.4.8

yanquanfahei committed:
style: typo (#4948)

DesignHhuang committed:
fix: the route path did not synchronize with the page (#4947)

mynetfan committed:
feat: add `confirmDisabled` for `Dialog` (#4959)

mynetfan committed:
feat: add `placement` for `Drawer` (#4958)

vince292007 committed:
feat: upgrade vite version to 6.0.0 (#4961)

* chore: upgrade vite version to 6.0.0

* chore: update lock

jeff-fe committed:
fix: button-control page mistake (#4963)

* fix: button-control page mistake

按钮控制展示逻辑错误

* fix: button-control.vue button text

mynetfan committed:
chore: tailwind css icon example (#4969)

mynetfan committed:
fix: element hover style in dark theme (#4983)

richex-cn committed:
chore: update deprecated document link in .github/ISSUE_TEMPLATE (#4986)

imdap committed:
fix: when VxeTable toolbarConfig.refresh is enabled, it will carry incorrect parameters (#4980)

gee1k committed:
fix: 修复 Form Api 根据字段名移除表单项，字段取反了的问题 (#4971)

huangfe1 committed:
chore: update modal.vue (#4987)

loading时候 子组件禁用点击事件

Co-authored-by: Vben <ann.vben@gmail.com>

mynetfan committed:
docs: add form slots docs (#4992)

mynetfan committed:
fix: mixed menu layout in full content mode (#4990)

leizhiyou committed:
fix: form 表单不支持field.xxx.xx格式的defaultValue配置 (#4965)

* fix: form 表单不支持field.xxx.xx格式的defaultValue配置

* chore: 修复代码规范问题

anncwb committed:
chore: update deps (#4993)

anncwb committed:
chore: Revert "fix: form 表单不支持field.xxx.xx格式的defaultValue配置 (#4965)"

This reverts commliit 12f216c0e7395caca21112064f0734c8ded424b3.

mynetfan committed:
feat: improve `page` component (#5013)

* feat: `page` component support fixed header

* docs: `page`  component documentation

* docs: Improve `props` types of `page`

* docs: improve `fixedHeader` description of `page`

* fix: `page` header border color with fixedHeader

* feat: add `headerClass` for `Page`

mynetfan committed:
fix: app config support `.env.local` (#5012)

mynetfan committed:
feat: form component `IconPicker` (#5005)

mynetfan committed:
feat: v-loading support for `element plus` (#5008)

mynetfan committed:
fix: extra menu title follow locale change (#5006)

mynetfan committed:
fix: header left padding fixed (#5007)

mynetfan committed:
fix: `Page` header class in fixed mode (#5023)

anncwb committed:
feat: add api-select component (#5024)

anncwb committed:
chore: release 5.5.0

DesignHhuang committed:
fix: select Long option style problem (#5030)

DesignHhuang committed:
feat: add submitOnChange props to vben form (#5032)

mynetfan committed:
fix: `dialog` and `drawer` footer gap in small screen (#5025)

mynetfan committed:
chore: codeowners update (#5048)

mynetfan committed:
fix: replace input component in `IconPicker` (#5047)

* fix: replace input component in `IconPicker`

* chore: fixed IconPicker demo

mynetfan committed:
fix: modal radius not follow preferences (#5063)

mynetfan committed:
fix: `form` prop `handleValuesChange` no effect (#5060)

mynetfan committed:
fix: validate message not display, fix #5034 (#5038)

mynetfan committed:
fix: range picker props fixed for element-plus (#5042)

mynetfan committed:
fix: form setValues not support  `dayjs` and `Date` value (#5064)

* fix: setFormValues not support  `dayjs object` value

* fix: setFormValues not support `Date` value

* chore: remove console log

mynetfan committed:
fix: layout overflow style (#5066)

mynetfan committed:
chore: page prop type check (#5067)

mynetfan committed:
chore: remove useless `fixedHeader` prop for `Page` (#5069)

mynetfan committed:
feat: imporve naive form component (#5071)

mynetfan committed:
feat: improve element plus form component (#5072)

mynetfan committed:
feat: improve `ApiSelect` component (#5075)

* feat: improve `ApiSelect` component

* chore: `ApiSelect` props name changed

mynetfan committed:
feat: add default `placeholder` for `ApiSelect` (#5078)

mynetfan committed:
feat: modal&drawer support appendToMain and zIndex (#5092)

* feat: modal/drawer support append to main content

* feat: modal zIndex support

* fix: drawer prop define

* chore: type

* fix: modal/drawer position fixed while append to body

* docs: typo

* chore: add full-width drawer in content area

* chore: remove unnecessary class

mynetfan committed:
chore: remove redundant test code (#5094)

ArthurDarkstone committed:
feat: replace `ElSelect` with `ElSelectV2` in component adapter for butter performance (#5085)

mynetfan committed:
refactor: `ApiComponent` with docs (#5099)

* refactor:  `ApiComponent` with docs

* docs: remove invalid docs

* docs: remove duplicate prop docs

* docs: update `ApiComponent` docs

mynetfan committed:
fix: table auto height (#5101)

mynetfan committed:
docs: fix typos (#5105)

mynetfan committed:
chore: update deps (#5107)

mynetfan committed:
feat: form support reverse action buttons (#5108)

* feat: form support reverse action buttons

* fix: submit button class

mynetfan committed:
fix: resolve table toolbar error (#5109)

mynetfan committed:
feat: table search form visible control (#5121)

* feat: table search form visible control

* chore: fix docs and demo

* chore: type error fixed

anncwb committed:
chore: release 5.5.1

mynetfan committed:
fix: form submission not appropriate (#5126)

mynetfan committed:
fix: form support `disabledOnInputListener` (#5127)

* fix: form support `disabledOnInputListener`

* chore: docs update

mynetfan committed:
feat: tabbar support mouse wheel vertical (#5129)

* feat: tabbar support mouse wheel

* docs: add tabbar wheelable tips

* chore: resolve vitest test

mynetfan committed:
fix: element plus validate failed style (#5130)

* fix: element plus validate failed style

* fix: element plus textarea style

mynetfan committed:
fix: form component events bind (#5137)

* fix: from component events bind

* chore: update docs

* chore: default value and docs sync

mynetfan committed:
fix: vscode debug profile (#5140)

mynetfan committed:
fix: vxeGrid default sort data no effect in first query (#5141)

* fix: vxeGrid default sort data no effect in first query

* fix: query params lost

mynetfan committed:
fix: pinInput value synchronous update (#5142)

mynetfan committed:
feat: user-dropdown support `hover` trigger (#5143)

* feat: user-dropdown support `hover` trigger

* fix: modified type declaration

mynetfan committed:
fix: login expired modal z-index (#5145)

* fix: login expired modal z-index

* feat: support custom z-index

mynetfan committed:
fix: `disabledOnChangeListener` not work in form (#5146)

mynetfan committed:
feat: auto activate subMenu on select root menu (#5147)

* feat: auto activate subMenu on click root menu

* fix: prop name fixed

* chore: test and docs update

mynetfan committed:
feat: `autoActivateChild` support more layout mode (#5148)

mynetfan committed:
feat: add demo for modify menu badge data

mynetfan committed:
feat: improve code login demo (#5154)

* feat: add some method in formApi

* fix: VbenPinInput style with small screen

* chore: improve code login demo

mynetfan committed:
feat: form `colon` support (#5156)

mynetfan committed:
docs: fix docs-link and add `EllipsisText` docs (#5158)

* docs: fix docs-link and add `EllipsisText` docs

* fix: ellipsisText docs link

mynetfan committed:
fix: form auto submit no effect when showDefaultActions is false (#5163)

mynetfan committed:
feat: form compact mode support (#5165)

mynetfan committed:
fix: user homePath no effect sometimes (#5166)

mynetfan committed:
fix: sidebar style on focus (#5178)

mynetfan committed:
feat: page content class override (#5179)

chengazhen committed:
fix: resolve eslint errors as well as TS type errors (#5172)

* fix: remove TypeScript error suppression for missing types in Vue ESLint config

* fix: enhance application configuration with CSS options type support

Booshaw committed:
docs: fix typos (#5169)

lonewolfyx committed:
fix: remove the overlap caused by border-b (#5160)

mynetfan committed:
fix: sidebar header height (#5183)

mynetfan committed:
feat: add `resizable` and `ColPage` component (#5188)

* feat: add component resizable

* feat: component `ColPage` with demo

mynetfan committed:
fix: grid tools in toolbar config not working as expected (#5190)

mynetfan committed:
fix: vxeGrid top padding (#5193)

anncwb committed:
fix: build error (#5199)

mynetfan committed:
fix: grid form submit button locale switch (#5205)

mynetfan committed:
chore: add apiSelect remote search demo (#5246)

mynetfan committed:
chore: update deps

mynetfan committed:
feat: header menu align support (#5256)

* feat: header menu align support

* fix: typo

mynetfan committed:
chore: downgrade vue-tsc version

anncwb committed:
chore: release 5.5.2

mynetfan committed:
feat: header mixed layout (#5263)

* feat: new layout header-mixed

* fix: header-mixed layout update

* feat: layout preference update

* fix: extra menus follow layout setting

mynetfan committed:
fix: header-mixed layout side-menu active (#5265)

* fix: header-mixed layout side-menu active

* fix: config test

mynetfan committed:
feat: modal support destroy on close

mynetfan committed:
feat: drawer support `onOpened` & `onClosed`

mynetfan committed:
feat: drawer support destroy on close

mynetfan committed:
docs: update dialog and drawer docs

mynetfan committed:
feat: drawer close icon placement (#5269)

mynetfan committed:
feat: new layout `sidebar nav with full header` (#5270)

mynetfan committed:
fix: header logo may not be collapsed in `header-sidebar-nav` layout (#5274)

mynetfan committed:
fix: breadcrumb setting not valid for `header-sidebar-nav` layout (#5275)

mynetfan committed:
fix: sidebar preferences fixed (#5276)

anncwb committed:
perf: format code with better style (#5283)

mynetfan committed:
fix: vben select placeholder color (#5286)

f2ngwx committed:
chore: update quick-start.md (#5303)

change COREPACK_REGISTRY to COREPACK_NPM_REGISTRY

clddup committed:
feat: useEcharts exports echarts instance#5294 (#5299)

wangwenting-dev committed:
fix: download from url triggered twice sometimes (#5319)

解决Chrome、Safari通过路径一次下载两个文件的BUG

mynetfan committed:
fix: form `fieldMappingTime` is not working (#5333)

* fix: form option `fieldMappingTime` is not working

* fix: form merge support `fieldMappingTime`

mynetfan committed:
fix: code lint

mynetfan committed:
fix: form `fieldMappingTime` improve and `modelPropName` support (#5335)

* 表单的fieldMappingTime支持将格式化掩码设为null以便原值映射，这样可以支持非日期时间类型的组件；
* 表单增加modelPropName设置组件的双向绑定属性名，用于支持未提前注册的双向绑定属性为非默认名称的组件。
* 增加一些经常会有人提到的组合字段演示，

mynetfan committed:
fix: form valid-error style in naive (#5336)

mynetfan committed:
fix: primaryColor calculation (#5337)

mynetfan committed:
chore: update deps

mynetfan committed:
fix: vxeGrid init without search form (#5342)

* 修复vxeGrid在未使用表单的情况下，自动加载数据失效的问题。

* 暂时将vxeGrid版本锁定在4.10.0，新版本尺寸计算尚有问题

mynetfan committed:
fix: locale switching logic correction (#5344)

* 修复语言切换后的数据更新逻辑

* 表单默认按钮的content属性可提供computed类型的值

mynetfan committed:
feat: allow close tab when mouse middle button click (#5347)

* 偏好设置增加鼠标中键关闭标签页的设置

mynetfan committed:
perf: modal and drawer api support chain calls (#5351)

* perf: modal and drawer api support chain calls

* fix: typo

mynetfan committed:
perf: add nested modal demo (#5353)

mynetfan committed:
feat: integrate new component `Tippy` with demo (#5355)

* 添加新的工具提示组件Tippy

mynetfan committed:
feat: improve `tippy` demo (#5357)

mynetfan committed:
feat: popup component support overlay blur effect (#5359)

mynetfan committed:
fix: useEcharts return invalid instance (#5360)

mynetfan committed:
ci: retry deploy while faild

mynetfan committed:
style: popover bgColor is too close to common (#5364)

修复Dark主题下，弹出层的背景色与主体背景色太过接近的问题

mynetfan committed:
fix: spinner may stop playing animation after dismiss (#5365)

* fix: spinner may stop playing animation after dismiss

* fix: animation paused more safely

mynetfan committed:
feat: add `noBasicLayout` in route meta (#5386)

所有菜单数据无需配置component为BasicLayout，它们将会默认使用基础布局，也可以通过meta.noBasicLayout来阻止这一行为

mynetfan committed:
fix: root router config fixed (#5389)

mynetfan committed:
feat: support set default props for drawer and modal (#5390)

* feat: support set default props for drawer and modal

* docs: fix typo

mynetfan committed:
fix: hide root route in breadcrumb

mynetfan committed:
perf: improve fieldMappingTime to support format function (#5392)

mynetfan committed:
style: element plus loading style fixed (#5393)

* Element Plus的loading组件默认zIndex太高

mynetfan committed:
refactor: fix popup component zIndex (#5397)

mynetfan committed:
feat: modal state locked on submitting (#5401)

* feat: modal state locked on submitting

* docs: 更新modal文档

mynetfan committed:
fix: demos route fixed (#5405)

mynetfan committed:
fix: form update state error before form mounted (#5406)

mynetfan committed:
docs: update docs (#5408)

mynetfan committed:
fix: mouse events ignored on modal loading (#5409)

mynetfan committed:
fix: antd button icon style (#5421)

mynetfan committed:
perf: request support to set how to return response (#5436)

* feat: request support to set how to return response

* docs: typo

* fix: test unit

* test: add request responseReturn test

mynetfan committed:
perf: easy to define fieldName of response data (#5442)

mynetfan committed:
feat: auto fetch icon list in iconPicker (#5446)

* feat: auto fetch icon list in iconPicker

* fix: add timeout controller for fetching

* feat: add pending controller

* fix: icon demo prefix

anncwb committed:
chore: release 5.5.3

mynetfan committed:
fix: request download and upload not support `responseReturn` (#5456)

* fix: request download and upload not support `responseReturn`

* docs: update

* fix: type of request client upload result

mynetfan committed:
fix: menu data for backend mode fixed (#5465)

mynetfan committed:
fix: renderComponentContent lose slot props data (#5466)

* 修复FormItem传递插槽时丢失插槽props的问题

jsxzhub committed:
docs: update request and access docs (#5468)

* fix: Update server.md

* docs: update request and access docs

mynetfan committed:
fix: ant tag icon default style (#5473)

mynetfan committed:
chore: remove testing code

mynetfan committed:
chore: update deps

mynetfan committed:
docs: update form docs (#5485)

mynetfan committed:
feat: tabbar support max count limit (#5490)

* 标签栏支持限制打开的最大数量

mynetfan committed:
fix: width for ellipsisText tooltip in popover content (#5517)

* 修复省略文本用在气泡中时，提示弹出层的宽度计算有误的问题

mynetfan committed:
fix: userDropdown triggered unnecessary while overlay shown (#5520)

* 修复顶部的用户资料下拉在弹窗被打开时，仍然可以被触发的问题

jinmao88 committed:
chore: add qq group 5 (#5530)

mynetfan committed:
feat: new component jsonViewer (#5544)

* 添加新组件JsonViewer用于展示JSON结构数据

mynetfan committed:
chore: update deps

mynetfan committed:
fix: auth api definition

mynetfan committed:
style: code style fixed

mynetfan committed:
feat: add more event for `jsonViewer` (#5546)

* 为JsonViewer添加事件支持

mynetfan committed:
fix: nitro server cors support with cookie (#5549)

* 修复nitro server在使用cookie时的跨域配置

mynetfan committed:
feat: demo of motion plugin (#5550)

添加Motion的用法例子

mynetfan committed:
fix: playground route missing

mynetfan committed:
refactor: new `CountTo` component with demo (#5551)

mynetfan committed:
fix: form item slot context fixed (#5552)

* 修复表单插槽

mynetfan committed:
chore: eslint rules update

anyup committed:
fix: solve the problem of inconsistent returns of formSchema custom field names when code login (#5563)

mynetfan committed:
fix: form item overflow fixed and layout improved (#5572)

* fix: form item overflow fixed and layout improved

* fix: basic form demo update

* feat: form label support render

* fix: form docs update

mynetfan committed:
fix: vben count to animator event name fixed (#5573)

fzq1998 committed:
fix: `CountTo` component resolve `separator` prop not taking effect (#5578)

mynetfan committed:
fix: form label and control style (#5580)

* fix: form label and control style

* fix: empty label mark with required rules

mynetfan committed:
feat: improve form demo (#5582)

mynetfan committed:
feat: loading and spinner component with directive (#5587)

* 添加loading和spinner组件，以及对应的vue指令

mynetfan committed:
fix: loading and spinner style fixed and improved (#5588)

mynetfan committed:
feat: add `VbenButtonGroup` and  `VbenCheckButtonGroup` with demo (#5591)

* 添加按钮组、选择按钮组以及相应的Demo

mynetfan committed:
fix: `search` take no effect in `icon-picker`  with antd (#5592)

littlesparklet committed:
fix: Fix inconsistent spacing around search form (issue #5429) (#5495)

mynetfan committed:
chore: output console error for invalid route component (#5593)

mynetfan committed:
chore: demo page for system/department (#5611)

* feat: department management demo

* perf: department page improve

* feat: demo api middleware

* fix: add losing import

mynetfan committed:
fix: getPopupContainer will return closet form first (#5612)

mynetfan committed:
chore: remove sleep in department list api

mynetfan committed:
chore: demo page menu management (#5619)

* 添加菜单管理演示页面

jinmao88 committed:
fix: useDrawer中closeIconPlacement设置无效 (#5624)

mynetfan committed:
docs: fix zod rules docs

mynetfan committed:
fix: hideChildrenInMenu demo code (#5626)

mynetfan committed:
fix: breadcrumb style is affected by the globally-imported antd styles (#5627)

* 修复全局引入Antd时，面包屑的样式会受到影响的问题

mynetfan committed:
fix: drawer close icon placement default value

mynetfan committed:
chore: update depts

mynetfan committed:
fix: logo style in login page is affected by the globally-imported antd styles

* 修复登录页左上角LOGO部分的文字在全局导入antd样式的时候位置不正确的问题

mynetfan committed:
docs: `vbenVxeTable` slots docs update

jasonz18 committed:
docs: typo

mynetfan committed:
feat: modal and drawer locking improve (#5648)

* feat: add `unlock` for modalApi

* fix: modal's close button style in locking

* fix: fix modal's close button disabled on locking

* feat: add `lock` and `unlock` for drawerApi

mynetfan committed:
fix: nitro server cookie maxAge fixed

mynetfan committed:
feat: vben checkbox support indeterminate state and transition animation (#5662)

mynetfan committed:
fix: keepAlive not working for popup `appendToMain` (#5666)

* 修复弹窗和抽屉 `appendToMain` 时且启用`keepAlive` 时未能正确缓存的问题

mynetfan committed:
fix: demo nested menu path (#5667)

* 修复演示的嵌套菜单path配置导致的面包屑跳转问题

mynetfan committed:
feat: role management page with component `tree` (#5675)

* feat: add shadcn tree

* fix: update vbenTree component

* feat: role management demo page

* feat: add cellSwitch renderer for vxeTable

* chore: remove tree examples

mynetfan committed:
fix: vxeTable search button not working with slot (#5678)

mynetfan committed:
feat: add size prop to avatar component and update logo component for size handling (#5684)

ijackwu committed:
fix: live-server set port use [--port=PORT] (#5687)

mynetfan committed:
chore: update depts

mynetfan committed:
fix: improve component repackaging

mynetfan committed:
feat: improved formApi for component instance support

* 改进表单API以支持组件实例的获取，以及焦点字段的获取

mynetfan committed:
docs: update table slots docs

mynetfan committed:
feat: auto set component name for keep-alive (#5690)

* fix: auto set component name for keep-alive

* fix: type define

mynetfan committed:
fix: miss default value in vbenLayout

mynetfan committed:
fix: form item style adjustment (#5694)

chen-d-yu committed:
fix: vsh stylelint script fixed (#5729)

anncwb committed:
chore: release 5.5.4

anyup committed:
fix: sort the menu so that it doesn't get replaced with 999 when order=0 (#5753)

Co-authored-by: anyup <anyupxing@163.com>

mynetfan committed:
fix: table form reset will trigger reload twice while submitOnChange set (#5756)

* 修复表格的搜索表单在重置时，可能会触发2次表格刷新的问题

jinmao88 committed:
chore: update codeowners (#5750)

mynetfan committed:
feat: add `form-is-required` class for required items. fixed: #5739 (#5759)

mynetfan committed:
fix: base component focus color in form (#5760)

mynetfan committed:
fix: default value for nested fields (#5763)

jinmao88 committed:
feat(ui): logo icon support click events (#5725)

* feat(ui): 扩展auth页面添加点击 Logo 的事件处理

在 `authentication.vue` 中新增 `clickLogo` 属性，允许在点击 Logo 时执行自定义操作。在 `auth.vue` 中实现了一个示例的点击事件处理函数，用于测试该功能。

* feat(layout): 添加点击 logo 的事件处理函数

在 BasicLayout 组件中添加了 clickLogo 事件处理函数，并通过 emit 方法触发 clickLogo 事件。同时，在 basic.vue 中实现了 handleClickLogo 函数，用于处理 logo 点击事件。

* fix(ui): 移除logo点击事件的控制台日志

mynetfan committed:
fix: menu drawer can not be closed after a successful submitting (#5770)

zwtvip committed:
perf: 优化多文件上传入参是数组的情况 (#5757)

Co-authored-by: Jin Mao <50581550+jinmao88@users.noreply.github.com>

mynetfan committed:
fix: popover background color in dark mode (#5783)

* 修复dark主题下的弹出层背景色在某些浏览器上表现为完全透明的问题

mynetfan committed:
fix: auto check parent after node selected (#5794)

mynetfan committed:
perf: improve component packaging to enable instance method retrieval (#5795)

改进组件适配器里的包装函数，使得组件暴露的方法可以透传

mynetfan committed:
perf: improve api-component for using in form (#5796)

mynetfan committed:
fix: vben tree component warning (#5809)

mynetfan committed:
docs: fix form `compact` docs (#5811)

* docs: fix form `compact` docs

* docs: remove `compact` from FormCommonConfig

mynetfan committed:
feat: pre-set serialization methods for request parameters (#5814)

添加快捷设置请求参数序列化方法的配置

zwtvip committed:
style: 更正引用格式 (#5784)

anyup committed:
fix: route root.children duplicate problem (#5755)

Co-authored-by: anyup <anyupxing@163.com>

jinmao88 committed:
feat(tabbar): 添加右键菜单过滤功能 (#5820)

mynetfan committed:
fix:  watermark settings in the preferences modified accidentally (#5823)

jinmao88 committed:
feat: sidebar button config (#5818)

* feat: 新增 PreferenceCheckboxItem 组件

* feat(preferences): 添加侧边栏按钮配置功能

* feat: 新增按钮点击事件触发功能

* feat(SidebarPreferences): 新增侧边栏折叠按钮与固定按钮配置

* feat(ui): 新增侧边栏固定按钮及配置选项

* fix(test): 修正侧边栏配置项缺失问题

mynetfan committed:
fix: vbenTree modelValue synchronization (#5825)

mynetfan committed:
chore: add demo for apiComponent with caching and concurrency (#5827)

* chore: add demo for apiComponent with caching and concurrency

* docs: update api component docs

mynetfan committed:
docs: fix api-component demo link (#5828)

wu-clan committed:
Merge branch 'main' into create_vben

wu-clan committed:
feat: create vben project

wu-clan committed:
feat: adapt the login interface
